www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+
+The first thing that you'll want to do is install React in your project. To do that, run the following command in your project directory:
+
+To install these you can run:
+
+```bash
+npm install react react-dom --save
+```
+
+> In this tutorial commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+React uses a syntax called [JSX](https://reactjs.org/docs/introducing-jsx.html) for components. Snowpack has built in support for JSX files with the extension `.jsx`. To create your first React component in Snowpack rename `index.js` to `index.jsx`.
+
+> You may have used JSX in `.js` files before, but Snowpack expects JSX to be .jsx files
+
+```bash
+mv index.js index.jsx
+```
+
+> Remember that your index.html file is importing `/index.js`, not `/index.js`. Do you need to update it to match the new file? You don't because Snowpack turns `index.jsx` (which wouldn't run in the browser) into the `index.js` (which would).
+
+You can now import React into `index.jsx` and render a simple test component just to make sure it's working.
+
+Import React and add this test component `index.jsx` so you can use it.
+
+```diff
+/* Add JavaScript code here! */
+- console.log('Hello World! You did! Welcome to Snowpack :D');
++import React from 'react';
++import ReactDOM from 'react-dom';
++ReactDOM.render(<div>"HELLO REACT"</div>, document.getElementById('root'));
+```
+
+If you run this code you'll get `Error: Target container is not a DOM element.` The call to `document.getElementById('root')` is failing because the HTML page has no element with that ID. Head to index.html and add one in the `<body>`
+
+```diff
+  <body>
+-    <h1>Welcome to Snowpack!</h1>
++    <div id="root"></div>
+    <script type="module" src="/index.js"></script>
+```
+
+Now it should run and you should see "HELLO REACT"
+
+Image: screenshot of the project, which shows "HELLO REACT" on a white background
+
+## Organizing your files
+
+If you're going to be adding a bunch of new files, you probably don't want them in the root directory. A standard way of organizing React projects is an `src` directory for components and their associated files. Then a public directory for static assets like images, fonts, icons, and sometimes CSS. To do this you'll need to reorganize your files, then tell Snowpack about it in the configuration file, and update your paths in index.html.
+
+```
+📁 src
+↳ index.jsx
+📁 public
+↳ index.css
+↳ index.html
+```
+
+```bash
+mkdir src
+mkdir public
+mv index.jsx src/index.jsx
+mv index.html public/index.html
+mv index.css public/index.css
+```
+
+You need to tell Snowpack about this new structure using the [Snowpack Configuration API](https://www.snowpack.dev/#configuration) in the `snowpack.config.js` file. Right now it's a bare bones scaffold that doesn't do anything. Add this to the `mount` object:
```

Comment:
Good idea to mention the API, but not so fast. As a reader, I'd click it immediately and then get overwhelmed. How about:

Once your new structure is in place, you need to tell Snowpack about it. Every Snowpack project comes with a `snowpack.config.js` file for any configuration that you might need. Right now, you should see a mostly-empty, bare-bones config file. Add this to the empty `mount` object:

```
{CODE SNIPPET}
```

In the section below, you can link to the full API reference